src/configurations/cedict.js

```diff
@@ -0,0 +1,134 @@
+        name: 'meta',
+        primaryIndex: {
+          auto: true
+        }
+      },
+
+      /*
+      This is a store that keeps dictionary entries themselves.
+       */
+      dictionary: {
+        name: 'dictionary',
+        primaryIndex: {
+          /*
+          What property of a dictionary entry will become a primary index.
+           */
+          keyPath: 'index'
+        },
+
+        /*
+        The following defines the secondary indexes. The name of an index is used to address it
+        during queries. keyPath defines what prop of a dictionary entry will be used to build an index.
+         */
+        indexes: {
+          traditional: {
+            name: 'traditionalHwIdx',
+            keyPath: 'traditionalHeadword',
+            unique: false
+          },
+          simplified: {
+            name: 'simplifiedHwIdx',
+            keyPath: 'simplifiedHeadword',
+            unique: false
+          }
+        },
+        volatileStorage: {
+          /*
+          If volatile storage is disabled, all queries will run against an IndexedDB. This will minimize
+          RAM usage and is fast enough for most purposes (from several to tens of milliseconds). Enabling
+          volatile storage will place data into RAM and data will be retrieved faster at cost of a higher
+          RAM usage.
+           */
+          enabled: false,
+
+          /*
+          If volatile storage is indexed it will create additional in-memory maps to store headword indexes.
+          It will result in almost instantaneous retrieval of data at cost of a higher RAM usage.
+           */
+          indexed: false
+        },
+        permanentStorage: {
+          /*
+          With permanents storage enabled all CEDICT data will be saved into an IndexedDB and will stay there
+          between page reloads. This will allow not to download all CEDICT data each time the CEDICT service
+          is started. It will increase a service start time significantly (by tens of seconds, usually).It
+          will also spare several megabytes of network traffic.
+
+          With permanent storage enabled clients will be able to run searches directly against an IndexedDB
+          thus keeping RAM usage at a minimum.
+
+          It is highly recommended to have permanent storage always enabled except for cases when
+          a target device does not support it.
+           */
+          enabled: true,
+
+          /*
+          (Currently not implemented.)
+          Disabling permanent store indexes will slow searches down significantly (up to more than a second).
+          On the other hand, having indexes enabled to not increase IndexedDB size significantly.
+          Because of that it is recommended to always have this option on.
+           */
+          indexed: true
+        }
+      }
+    }
+  },
+
+  /*
+  Describes CEDICT data on a remote server that is required to run the current version of CEDICT service.
+   */
+  data: {
```

Comment:
Every time we roll out new CEDICT data changes we're supposed to update this file and rebuild the service. Thus we'll explicitly declare that the service is aware of the new data is ready to support it. Unless this is done, CEDICT client will use old data. That will add a safety layer as we'll be able to add a new CEDICT data to the server and test it without our clients using it (it CEDICT service is not rebuilt it will use old data even if the new one is available). The new data can be tested at the same time using a test version of CEDICT client (not available to general users).

Since CEDICT data and service script reside at the same place and will be updated simultaneously clients will receive a synchronized version of service and data instantaneously, after a page reload. That shall prevent any incompatibility issues between a data and a script.

What do you think? Does such policies make sense? Updating CEDICT metadata in this file and rebuilding a service script can be routinely made via an automated server-side process but we can resort to manual updates in cases of more complex data changes.

This script can also be converted to JSON. Would it make things cleaner?

Reply:
We can also split `storage` and `data` branches into separate files. Would it make things more flexible?

Reply:
> It makes more sense to me as a JSON file but I don't feel too strongly about it.

I mostly put it into a JS file to have an ability to add commends 🙂. But if to use JSON schema as you suggest we can put comments to the schema and the problem will be solved. So I think moving to JSON (in the next iteration probably) can be a good move. Would that be fine with you?

Reply:
>So I think moving to JSON (in the next iteration probably) can be a good move. Would that be fine with you?
yes